pkg/broker/server/server.go

```diff
@@ -149,35 +151,93 @@ func (s *server) createServiceInstance(w http.ResponseWriter, r *http.Request) {
 		req.Parameters = make(map[string]interface{})
 	}
 
-	if result, err := s.controller.CreateServiceInstance(id, &req); err == nil {
-		if result.Code == 0 {
-			result.Code = http.StatusCreated
-		}
-		util.WriteResponse(w, result.Code, result)
-	} else {
-		if result.Code == 0 {
-			result.Code = http.StatusBadRequest
-		}
 
-		util.WriteResponse(w, result.Code, &brokerapi.BrokerResponseError{
-			Code:        result.Code,
-			Description: err.Error(),
-		})
+	q := r.URL.Query()
+	acceptsIncomplete := q.Get("accepts_incomplete")
+	async := acceptsIncomplete == "true"
+	if acceptsIncomplete == "" || async != true {
```

Comment:
do we need both checks here? Could it be simplified to ``` if ! asnyc```

Reply:
Yeah I think it can be :+1: